@here/harp-examples/src/threejs_shadows.ts

```diff
@@ -189,22 +183,11 @@ const addOmvDataSource = (): Promise<void> => {
     return map.addDataSource(omvDataSource);
 };
 
-const patchFillStyle = (styleDeclaration: StyleDeclaration) => {
-    if (!isJsonExpr(styleDeclaration)) {
-        const style = styleDeclaration as Style;
-        if (style.technique === "fill") {
-            (style as any).technique = "standard";
-        }
-    }
-};
-
 /**
- * Replace all occurences of "fill" technique in the theme with "standard" technique.
- * "standard" technique is using three.js MeshStandardMaterial and is needed to receive
- * shadows.
+ * Replaces the lights in the style to cast shadows.
  * @param theme The theme to patch
  */
-const patchTheme = (theme: Theme) => {
+const fixLights = (theme: Theme) => {
```

Comment:
We don't have to patch loaded theme like this. We can simplify this by using builtin theme inheritance.

You can create right theme using `extends` clause right from beginning.

Just initialize MapView like thiscreate your theme like this:
```
const theme: Theme = { 
   extends: "resources/berlin_tilezen_base.json",
   lights: [... desired definitions of lights ]
}
initializeMapView(..., theme)
```

Much cleaners and encourages to use "preferred" methods for composing themes. 

Of course it wasn't completly possible in previous patch because we don't support "technique.name" rewrite.

Reply:
Thanks a lot! Good to know.